utils/src/main/scala/it/pagopa/interop/commons/utils/errors/AkkaResponses.scala

```diff
@@ -11,32 +11,35 @@ import it.pagopa.interop.commons.utils.errors.GenericComponentErrors.GenericErro
 
 trait AkkaResponses {
 
-  private def completeWithError(statusCode: StatusCode, error: ComponentError)(implicit
-    contexts: Seq[(String, String)],
-    serviceCode: ServiceCode
-  ): StandardRoute = complete(statusCode.intValue, Problem(statusCode, error, serviceCode, getCorrelationId(contexts)))
-
-  private def completeWithErrors(statusCode: StatusCode, errors: List[ComponentError])(implicit
-    contexts: Seq[(String, String)],
-    serviceCode: ServiceCode
-  ): StandardRoute = complete(statusCode.intValue, Problem(statusCode, errors, serviceCode, getCorrelationId(contexts)))
-
   private def completeWithError(statusCode: StatusCode, headers: List[HttpHeader], error: ComponentError)(implicit
     contexts: Seq[(String, String)],
     serviceCode: ServiceCode
-  ): StandardRoute =
-    complete(statusCode.intValue, headers, Problem(statusCode, error, serviceCode, getCorrelationId(contexts)))
+  ): StandardRoute = headers match {
+    case Nil => complete(statusCode.intValue, Problem(statusCode, error, serviceCode, getCorrelationId(contexts)))
+    case _   =>
+      complete(statusCode.intValue, headers, Problem(statusCode, error, serviceCode, getCorrelationId(contexts)))
+  }
+
+  private def completeWithErrors(
+    statusCode: StatusCode,
+    headers: List[HttpHeader],
+    errors: List[ComponentError]
+  )(implicit contexts: Seq[(String, String)], serviceCode: ServiceCode): StandardRoute = headers match {
+    case Nil => complete(statusCode.intValue, Problem(statusCode, errors, serviceCode, getCorrelationId(contexts)))
+    case _   =>
+      complete(statusCode.intValue, headers, Problem(statusCode, errors, serviceCode, getCorrelationId(contexts)))
+  }
 
   @inline private def getCorrelationId(contexts: Seq[(String, String)]): Option[String] =
     contexts.collectFirst { case (k, v) if k == CORRELATION_ID_HEADER => v }
 
-  def badRequest(error: ComponentError, logMessage: String)(implicit
+  def badRequest(error: ComponentError, logMessage: String, headers: List[HttpHeader] = Nil)(implicit
```

Comment:
optional parameters often cause headache, I would duplicate all functions and make the ones without the `headers` parameter call the other.
To be applied to all functions.

Eg for `badRequest`
```
  def badRequest(error: ComponentError, logMessage: String)(implicit
    contexts: Seq[(String, String)],
    logger: LoggerTakingImplicit[ContextFieldsToLog],
    serviceCode: ServiceCode
  ): StandardRoute = badRequest(error, logMessage, Nil)

  def badRequest(error: ComponentError, logMessage: String, headers: List[HttpHeader])(implicit
    contexts: Seq[(String, String)],
    logger: LoggerTakingImplicit[ContextFieldsToLog],
    serviceCode: ServiceCode
  ): StandardRoute = {
    logger.warn(logMessage, error)
    completeWithError(StatusCodes.BadRequest, headers, error)
  }
```